junit5/src/main/java/org/jboss/weld/junit5/WeldCustomizer.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.weld.junit5;
+
+import org.jboss.weld.environment.se.Weld;
+
+/**
+ * If no {@link WeldInitiator} field annotated with {@link WeldSetup} is present on a test class, all service providers of this interface are used to customize
+ * the "default deployment". The initial {@link Weld} instance is created using {@link WeldInitiator#createWeld()}.
+ *
+ * <p>
+ * A system property with key equal to FQCN of a customizer class may be used to disable a customizer completely. E.g. for a class
+ * {@code org.weld.FooWeldCustomizer} use {@code -Dorg.weld.FooWeldCustomizer=false} to disable the customizer.
+ * </p>
+ *
+ * @author Martin Kouba
+ */
+public interface WeldCustomizer {
```

Comment:
I still don't like this name but I am struggling to come up with better one.
`WeldBootstrapStrategy`, `WeldStartupDefinition`, `WeldBootstrapDefinition` - something along these lines per chance?

Reply:
I think this SPI is nor a strategy nor a definition. It's a contract for components which would customize/modify the default configuration. Also I'd like to avoid using the words like extension, configurator and builder.

Reply:
> Also I'd like to avoid using the words like extension, configurator and builder.

Agreed, I avoided them purposely too.

I would prefer some naming which underlines that you are modifying how Weld bootstraps. Customizer is just too general in my opinion although I can see why you picked that name.